### ISSUE TITLE
Support for passing std::unique_ptr<T> objects to chaiscript

### ISSUE THREAD

**Issue by jagoly** [ChaiScript/ChaiScript#215]
Currently, ChaiScript allows automatic conversions between pointer, reference, shared_ptr, reference_wrapper and value types. However, support for std::unique_ptr is currently missing. It would be useful to allow unique_ptrs to be treated the same way as other pointer/reference types.

see http://discourse.chaiscript.com/t/adding-unique-ptr-members-to-classes/116

**Comment by lefticus** [comment]
Fixed in fb7f8f194cc85daec881605e62bef85607a2ff66

**Comment by asyniakov** [comment]
Hello @lefticus,
```

chai.add(chaiscript::fun(&Document::addObject), "addObject");

class Document {
public:
    void addObject(std::unique_ptr<ObjectInterface>&& object) {
     ...
   }
};
```

doesn't work for me, but the free function works fine. Is it still non-working?
